common/diagnostics/Metrics.java

```diff
@@ -0,0 +1,221 @@
+    private final SystemProperties system;
+    private final NetworkRequests requests;
+    private final UsageCounters usage;
+    private final UserErrorStatistics userErrors;
+
+    public Metrics(String serverID, String name, String version) {
+        this.system = new SystemProperties(serverID, name, version);
+        this.requests = new NetworkRequests();
+        this.usage = new UsageCounters();
+        this.userErrors = new UserErrorStatistics();
+    }
+
+    public void requestAttempt(Metrics.NetworkRequests.Kind kind) {
+        requests.attempt(kind);
+    }
+
+    public void requestSuccess(Metrics.NetworkRequests.Kind kind) {
+        requests.success(kind);
+    }
+
+    public void setCurrentCount(UsageCounters.Kind kind, long value) {
+        usage.set(kind, value);
+    }
+
+    public void registerError(String errorCode) {
+        userErrors.register(errorCode);
+    }
+
+    protected String formatPrometheus() {
+        return String.join("\n", system.formatPrometheus(), requests.formatPrometheus(), usage.formatPrometheus(), userErrors.formatPrometheus());
+    }
+
+    protected JsonObject asJSON() {
+        JsonObject metrics = new JsonObject();
+        metrics.add("system", system.asJSON());
+        metrics.add("requests", requests.asJSON());
+        metrics.add("DB usage", usage.asJSON());
+        metrics.add("user errors", userErrors.asJSON());
+        return metrics;
+    }
+
+    protected String formatJSON() {
+        return asJSON().toString();
+    }
+
+    static class SystemProperties {
+        private final String serverID;
+        private final String name;
+        private final String version;
+
+        SystemProperties(String serverID, String name, String version) {
+            this.serverID = serverID;
+            this.name = name;
+            this.version = version;
+        }
+
+        JsonObject asJSON() {
+            JsonObject system = new JsonObject();
+            system.add("TypeDB version", name + " " + version);
+            system.add("Server ID", serverID);
+            system.add("Time zone", TimeZone.getDefault().getID());
+            system.add("Java version", System.getProperty("java.vendor") + " " + System.getProperty("java.version"));
+            system.add("Platform", System.getProperty("os.name") + " " + System.getProperty("os.arch") + " " + System.getProperty("os.version"));
+            return system;
+        }
+
+        String formatPrometheus() {
+            return "# TypeDB version: " + name + " " + version + "\n" +
+                    // no serverID, that's for reporting only
+                    "# Time zone: " + TimeZone.getDefault() .getID() + "\n" +
+                    "# Java version: " + System.getProperty("java.vendor") + " " + System.getProperty("java.version") + "\n" +
+                    "# Platform: " + System.getProperty("os.name") + " " + System.getProperty("os.arch") + " " + System.getProperty("os.version") + "\n";
+        }
```

Comment:
Is it possible for these to error at any point? What will happen if so? (eg. maybe we should wrap in a try/catch so the server doesn't happe to crash in some envs?)

Reply:
Empty `getProperty()` is not an error, and `TimeZone.getDefault()` is never `null` (it's set to GMT as last resort), so `.getID()` should not be able to error either,